### DISCUSSION TITLE
Errors after LDAP integration

### DISCUSSION THREAD

**Discussion by slinky18** [netbox-community/netbox#8410] [Q&A]
I am installing Netbox on Ubuntu server 20.04 running on a VMware server. I had everything running fine. I could add local users, log on and make edits. I was trying to get LDAP integration to work and now when I load the page I get:

"Server Error
A module import error occurred during this request. Common causes include the following:

Missing required packages - This installation of NetBox might be missing one or more required Python packages. These packages are listed in requirements.txt and local_requirements.txt, and are normally installed as part of the installation or upgrade process. To verify installed packages, run pip freeze from the console and compare the output to the list of required packages.

WSGI service not restarted after upgrade - If this installation has recently been upgraded, check that the WSGI service (e.g. gunicorn or uWSGI) has been restarted. This ensures that the new code is running.

The complete exception is provided below:

<class 'ModuleNotFoundError'>

No module named '10'

Python version: 3.8.10
NetBox version: 3.1.6"

I have checked all of the services in requirements.txt and local_requirements.txt using pip freeze. They are all there. The only problem I know I had during LDAP setup was with:

"source /opt/netbox/venv/bin/activate
pip3 install django-auth-ldap"

When I run source pip3 install django-auth-ldap, I would get a permissions error so I ran it as sudo pip3 install django-auth-ldap. Searching through I haven't seen anyone else with the "No module named '10'" error. Any tips on what to look for? I am relatively new to Linux so you may have to explain it like I am 5.

**Comment by Senetus** [comment]
Try to run sudo pip3 install django-auth-ldap --user instead. Maybe that helps to make it work for the netbox user too.

**Comment by DanSheps** [comment]
> When I run source pip3 install django-auth-ldap, I would get a permissions error so I ran it as sudo pip3 install django-auth-ldap. Searching through I haven't seen anyone else with the "No module named '10'" error. Any tips on what to look for? I am relatively new to Linux so you may have to explain it like I am 5.

You most likely installed the venv as a user not the user you are using.  You need to chown the directory appropriately to the netbox user and run the source and pip commands again.
